tensorboard/webapp/plugins/npmi/data_source/npmi_data_source.ts

```diff
@@ -124,23 +175,91 @@ export class NpmiHttpServerDataSource implements NpmiDataSource {
     );
   }
 
-  private fetchAnnotations() {
-    return this.http.get<AnnotationListing>(
-      this.httpPathPrefix + '/annotations'
+  private fetchAnnotations(experimentIds: string[]) {
+    const fetches = experimentIds.map((experimentId) => {
+      const url = `/experiment/${experimentId}/${this.httpPathPrefix}/annotations`;
```

Comment:
I would keep as-is `/experiment/${experimentId}/${this.httpPathPrefix}/annotations` to be consistent with the Time Series data source.  Relying on the backend to strip the experiment/:id is safe for now, as changing that behavior would break Time Series as well.